addon/components/discover-page/component.js

```diff
@@ -0,0 +1,808 @@
+                if (filter === 'providers') {
+                    restoreActiveFilters[filter] = this.get('theme.isProvider') ? this.get('activeFilters.providers') : [];
+                } else {
+                    restoreActiveFilters[filter] = [];
+                }
+
+            });
+            this.set('activeFilters', restoreActiveFilters);Ember.get(this, 'metrics')
+                .trackEvent({
+                    category: 'button',
+                    action: 'click',
+                    label: 'Discover - Clear Filters'
+                });
+        },
+        filtersChanged() {
+            // Ember SHARE action. Fired in faceted-search component when Ember-SHARE facets are modified.
+            this.search();
+        },
+        loadPage(newPage, scroll = true) {
+            if (newPage === this.get('page') || newPage < 1 || newPage > this.get('totalPages')) {
+                return;
+            }
+            this.set('page', newPage);
+            if (scroll) {
+                this.scrollToResults();
+            }
+            this.loadPage();
+        },
+        modifyRegistrationType(filter, query) {
+            // For REGISTRIES only - modifies "type" query param if "provider" query param changes.
+            // Registries are unusual, since the OSF Registration Type facet depends upon the Providers facet
+            if (filter === 'provider' && this.get('consumingService') === 'registries') {
+                if (query.length === 1 && query[0] === 'OSF') {
+                    this.set('type', this.get('activeFilters.types').join('OR'));
+                } else {
+                    this.set('type', '');
+                }
+            }
+        },
+        removeFilter(type, filterValue) {
+            // Ember-SHARE action.  Could be used to remove filters in Active Filters box (when Ember-SHARE and PREPRINTS/REGISTRIES code here is integrated)
+            let currentValue = getSplitParams(this.get(type)) || [];
+            let index = currentValue.indexOf(filterValue);
+            if (index > -1) {
+                currentValue.splice(index, 1);
+            }
+            currentValue = currentValue.length ? encodeParams(currentValue) : '';
+            this.set(type, currentValue);
+            this.get('facetFilters');
+        },
+        search() {
+            // Only want to track search here when button clicked. Keypress search tracking is debounced in trackSearch
+            Ember.get(this, 'metrics')
+                .trackEvent({
+                    category: 'button',
+                    action: 'click',
+                    label: 'Discover - Search',
+                    extra: this.get('q')
+                });
+
+            this.search();
+        },
+        selectSortOption(option) {
+            // Runs when sort option changed in dropdown
+            this.set('sort', option);
+            Ember.get(this, 'metrics')
```

Comment:
I just brought this over from preprints but my guess is that it was using same conventions from the ember-metrics docs https://github.com/poteto/ember-metrics#usage